File: JavaScript/src/bin/easy_108.js
/*
 * Daily Coding Problem: 108
 * Difficulty: Easy
 * Asked By: Google
 * Authored By: TenthEdict
 */

function check_string_rotation(a, b) {
    if (a.length != b.length) { return false };
    return (a + a).includes(b);
}

console.log(check_string_rotation("abcde", "cdeab"));
console.log(check_string_rotation("abcde", "abcde"));
console.log(check_string_rotation("abc", "acb"));
console.log(check_string_rotation("abcde", "abcd"));
console.log(check_string_rotation("", ""));
console.log(check_string_rotation("a", "a"));

File: Rust/src/bin/easy_108.rs
// Daily Coding Problem: 108
// Difficulty: Easy
// Asked by: Google
// Authored by: TenthEdict

fn check_string_rotation(a: &str, b: &str) -> bool {
    if a.len() != b.len() { return false }
    let doubled = a.repeat(2);
    doubled.contains(b)
}

fn main() {
    // Should be true
    println!("abcde, cdeab: {}", check_string_rotation("abcde", "cdeab")); // true
    println!("abcde, abcde: {}", check_string_rotation("abcde", "abcde")); // true (0 rotations)
    
    // Should be false
    println!("abc, acb: {}", check_string_rotation("abc", "acb"));   // false
    println!("abc, abcd: {}", check_string_rotation("abc", "abcd")); // false (different lengths)
    
    // Edge cases
    println!("'', '': {}", check_string_rotation("", ""));           // what do you expect?
    println!("a, a: {}", check_string_rotation("a", "a"));           // true
}
